job/job.go

```diff
@@ -34,6 +35,34 @@ type Job struct {
 	instance *packet.Instance
 }
 
+var tinkWorkflowClient = client.GetWorkflowsFromTink
```

Comment:
please no global variable, just pass in a client to `discoverActiveWorkflowForMac`

Reply:
My main objective of having this variable is to mock client.GetWorkflowsFromTink in unit test under job/job_test.go. If we want to remove this, Do you think that can be done in other way ?

Reply:
Yep change `discoverActiveWorkflowForMac(hwID)` to be `discoverActiveWorkflowForMac(client, hwID)` and then all the callers need to pass in what they want. Normal code can do `client.GetWorkflowsFromTink`, test code can do the mock.

Reply:
@mmlb i have added interfaces to mock `client.GetWorkflowsFromTink`. Can you please confirm if the changes are right